torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -168,186 +156,32 @@ def __init__(
         neighbor_sampler: Optional[NeighborSampler] = None,
         **kwargs,
     ):
-        # Remove for PyTorch Lightning:
-        if 'dataset' in kwargs:
-            del kwargs['dataset']
-        if 'collate_fn' in kwargs:
-            del kwargs['collate_fn']
-
-        self.data = data
+        # Get input type:
+        # TODO(manan): this computation is required twice, once here and once
+        # in LinkLoader:
+        edge_type, _ = get_edge_label_index(data, edge_label_index)
 
-        # Save for PyTorch Lightning < 1.6:
+        # Store num_neigbors to retain backwards compatibility:
```

Comment:
Why?

Reply:
Some examples explicitly queried `loader.num_neighbors`, so I left this in temporarily in case this was a more general pattern. If not, happy to remove.

Reply:
I think this is safe to remove. Let's mark it in the changelog as a potential breaking change.

Reply:
sgtm